### PR TITLE
Fix Unix socket subsystem

### PR DESCRIPTION
**Pull request by cgfandia-tii**
### Which kind of PR do you create?

- [x] This PR only contains minor fixes.
- [ ] This PR contains major feature update.
- [ ] This PR introduces a new function/api for Qiling Framework.

### Coding convention?

- [x] The new code conforms to Qiling Framework naming convention.
- [x] The imports are arranged properly.
- [x] Essential comments are added.
- [ ] The reference of the new code is pointed out.

### Extra tests?

- [ ] No extra tests are needed for this PR.
- [ ] I have added enough tests for this PR.
- [x] Tests will be added after some discussion and review.

### Changelog?

- [ ] This PR doesn't need to update Changelog.
- [x] Changelog will be updated after some proper review.
- [ ] Changelog has been updated in my PR.

### Target branch?

- [x] The target branch is dev branch.

### One last thing

- [x] I have read the [contribution guide](https://docs.qiling.io/en/latest/contribution/)

-----

### CODE REVIEW

File: qiling/os/posix/syscall/unistd.py [hunk 1]
```diff
@@ -143,12 +143,14 @@ def ql_syscall_faccessat(ql: Qiling, dfd: int, filename: int, mode: int):
 
     if not os.path.exists(real_path):
         regreturn = -1
-
-    elif stat.S_ISFIFO(Stat(real_path).st_mode):
+    else:
         regreturn = 0
+    # TODO: according to real implementation, idk why here was a FIFO check
```
Comment (wtdcode) [inline qiling/os/posix/syscall/unistd.py:148]: I track the history for this code. Looks like it's introduced here: https://github.com/qilingframework/qiling/commit/2f6ff4383ef7ae9b1707bdc5313f957849431ef3. Thus, it should be safe to remove the comments totatly.

File: qiling/os/posix/syscall/unistd.py [hunk 2]
```diff
@@ -732,7 +734,7 @@ def _type_mapping(ent):
         return bytes([t])
 
     if ql.os.fd[fd].tell() == 0:
-        n = ql.arch.pointersize
+        large_field_size = 8 if is_64 else 4
```
[the remark this answers is not in the document]
Comment (cgfandia-tii) [reply qiling/os/posix/syscall/unistd.py:737]: You are right, `large_field_size` ( `ino64_t` and `off64_t`) is `unsigned long long` on x64 - not pointers, so the size of the structure `linux_dirent64` does not depends on the architecture, but the initial implementation relies on the pointer size.

**Comment by cgfandia-tii** [comment]
Looks like during the `test_elf_linux_x8664_getdents` binary at `examples/rootfs/x8664_linux/bin/x8664_getdents` expects `linux_dirent64`, but `readdir` implementation of provided `libc.so` invokes `getdent` instead of `getdent64`
image

**Comment by elicn** [comment]
> Looks like during the `test_elf_linux_x8664_getdents` binary at `examples/rootfs/x8664_linux/bin/x8664_getdents` expects `linux_dirent64`, but `readdir` implementation of provided `libc.so` invokes `getdents` instead of `getdents64`.

That aligns with my assumption that `getdents64` or `getdents` might be called regardles of the arch used, and better rely on `arch.pointersize` rather than a flag coming from the caller. My guess is that reverting the changes made to the `getdents` implementation would solve this. I must admit that I didn't quite uderstand what had to be fixed there in the first place..

**Comment by cgfandia-tii** [comment]
> > Looks like during the `test_elf_linux_x8664_getdents` binary at `examples/rootfs/x8664_linux/bin/x8664_getdents` expects `linux_dirent64`, but `readdir` implementation of provided `libc.so` invokes `getdents` instead of `getdents64`.
> 
> That aligns with my assumption that `getdents64` or `getdents` might be called regardles of the arch used, and better rely on `arch.pointersize` rather than a flag coming from the caller. My guess is that reverting the changes made to the `getdents` implementation would solve this. I must admit that I didn't quite uderstand what had to be fixed there in the first place..

The cause of changing the implementation was a problem in the emulation of Android Armv7 binaries, where `getdents64` was called with `arch.pointersize == 4`, which leads to incompatibility of the `linux_dirent64` size. By the way, I will revert the changes, because I'm not sure that the problem in test binaries.

**Comment by elicn** [comment]
> The cause of changing the implementation was a problem in the emulation of Android Armv7 binaries, where `getdents64` was called with `arch.pointersize == 4`, which leads to incompatibility of the `linux_dirent64` size. By the way, I will revert the changes, because I'm not sure that the problem in test binaries.

Unless a malware sample, the binaries should be pretty standard so I'd assume they work properly. It's worth checking whether Android behaves differently then other POSIX here; if it does, then this method may be implemented specifically for Android: OS-specific syscalls implementation takes precesence over global POSIX syscalls'.

**Review by elicn — APPROVED** [review]

**Review by wtdcode — APPROVED** [review]
Looks good to me.

**Comment by wtdcode** [comment]
Once CI is green, we are safe to go!

**Pull request merged**